pro_tes/middleware/task_distribution/distance.py

```diff
@@ -37,18 +37,19 @@ def task_distribution(input_uri: List) -> List:
     Returns:
         A list of ranked TES instances, ordered by the minimum distance
         between the input files and each TES instance.
+
+    Raises:
+        ValueError: If no input URIs are available.
     """
     foca_conf = current_app.config.foca
     tes_uri: List[str] = foca_conf.tes["service_list"]
     access_uri_combination = get_uri_combination(input_uri, tes_uri)
 
+    if not input_uri:
+        raise ValueError("No input URIs available.")
+
```

Comment:
For performance reasons, this should come before anything else in this method. Especially because you are already using `input_uri` in the previous statement.

Reply:
Done